server.go

```diff
@@ -247,26 +247,30 @@ func newServer(listenAddrs []string, notifier chainntnfs.ChainNotifier,
 		return nil, err
 	}
 	for _, node := range linkNodes {
-		// Create a wrapper address which couples the IP and the pubkey
-		// so the brontide authenticated connection can be established.
-		lnAddr := &lnwire.NetAddress{
-			IdentityKey: node.IdentityPub,
-			Address:     node.Addresses[0],
-		}
 		pubStr := string(node.IdentityPub.SerializeCompressed())
-		srvrLog.Debugf("Attempting persistent connection to channel "+
-			"peer %v", lnAddr)
-
-		// Send the persistent connection request to the connection
-		// manager, saving the request itself so we can cancel/restart
-		// the process as needed.
-		// TODO(roasbeef): use default addr
-		connReq := &connmgr.ConnReq{
-			Addr:      lnAddr,
-			Permanent: true,
+		// In case a node has multiple addresses, attempt to connect to
+		// each of them.
+		for _, address := range node.Addresses {
+			// Create a wrapper address which couples the IP and the pubkey
+			// so the brontide authenticated connection can be established.
+			lnAddr := &lnwire.NetAddress{
+				IdentityKey: node.IdentityPub,
+				Address:     address,
+			}
+			srvrLog.Debugf("Attempting persistent connection to channel "+
+				"peer %v", lnAddr)
+
+			// Send the persistent connection request to the connection
+			// manager, saving the request itself so we can cancel/restart
+			// the process as needed.
+			// TODO(roasbeef): use default addr
+			connReq := &connmgr.ConnReq{
+				Addr:      lnAddr,
+				Permanent: true,
+			}
+			s.persistentConnReqs[pubStr] = connReq
+			go s.connMgr.Connect(connReq)
```

Comment:
Added a slice of connReqs to address issue #1. For #2, I believe @Roasbeef was going to add something in the connection manager to ensure that when a connection is established, the other connection attempts are cancelled. I believe this should prevent the [inbound|outbound]peerConnected from being called more than once, ensuring that duplicate peers won't be created.